src/Umbraco.Web/PublishedCache/IContentRouter.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Umbraco.Web.PublishedCache
+{
+    /// <summary>
+    /// Router for content ids
+    /// </summary>
+    public interface IContentRouter
+    {
+        /// <summary>
+        /// Gets content id identified by a route.
+        /// </summary>
+        /// <param name="preview">A value indicating whether to consider unpublished content.</param>
+        /// <param name="route">The route</param>
+        /// <param name="hideTopLevelNode">A value forcing the HideTopLevelNode setting.</param>
+        /// <returns>The content, or null.</returns>
+        /// <remarks>
+        /// <para>A valid route is either a simple path eg <c>/foo/bar/nil</c> or a root node id and a path, eg <c>123/foo/bar/nil</c>.</para>
+        /// <para>If <param name="hideTopLevelNode" /> is <c>null</c> then the settings value is used.</para>
+        /// <para>The value of <paramref name="preview"/> overrides defaults.</para>
+        /// </remarks>
+        ContentRoutingResult GetIdByRoute(IPublishedCache2 currentSnapshot,bool preview, string route, bool? hideTopLevelNode, string culture);
```

Comment:
IContentRouter is a singleton and so is `IPublishedCache2` so this can just be ctor injected?

Reply:
I don't think IPublishedCache2  is a singleton.

Reply:
yep you are correct